sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -22,14 +22,17 @@ import java.lang.{Boolean => JBool}
 import scala.collection.mutable.ArrayBuffer
 import scala.language.{existentials, implicitConversions}
 
+import org.apache.spark.sql.catalyst.trees.TreeNode
 import org.apache.spark.sql.types.{BooleanType, DataType}
 
 /**
  * Trait representing an opaque fragments of java code.
  */
-trait JavaCode {
+trait JavaCode extends TreeNode[JavaCode] {
```

Comment:
shall we only make `Block` extends `TreeNode`?

Reply:
I think it is ok. Currently `ExprValue` doesn't have to be `TreeNode`.

Reply:
Will update in next commit.